service/src/main/java/uk/gov/hmcts/reform/fpl/utils/CaseDetailsMap.java

```diff
@@ -13,12 +13,16 @@ private CaseDetailsMap(Map<? extends String, ?> map) {
     }
 
     public CaseDetailsMap putIfNotEmpty(String key, Object value) {
+        updateOrRemoveIfEmpty(this, key, value);
+        return this;
+    }
+
+    public static void updateOrRemoveIfEmpty(Map<String, Object> map, String key, Object value) {
```

Comment:
you probs don't want this to be static

Reply:
I would have the action method instead accept CaseDetailsMap then you don't need to update this